File: app/src/main/java/com/alekseyvalyakin/roleplaysystem/ribs/game/active/dice/diceresult/SingleDiceTypeResultViewModel.java
package com.alekseyvalyakin.roleplaysystem.ribs.game.active.dice.diceresult;

//public class SingleDiceTypeResultViewModel extends AbstractFlexibleItem<SingleDiceTypeResultViewHolder> {
//
//    private List<DiceResult> diceResults;
//    private int totalResultValue;
//    private DiceType diceType;
//    private boolean stateExpanded;
//
//    public SingleDiceTypeResultViewModel(List<DiceResult> diceResults, int totalResultValue, DiceType diceType, boolean stateExpanded) {
//        this.diceResults = diceResults;
//        this.totalResultValue = totalResultValue;
//        this.diceType = diceType;
//        this.stateExpanded = stateExpanded;
//    }
//
//    public boolean isStateExpanded() {
//        return stateExpanded;
//    }
//
//    public List<DiceResult> getDiceResults() {
//        return diceResults;
//    }
//
//    public int getTotalResultValue() {
//        return totalResultValue;
//    }
//
//    public DiceType getDiceType() {
//        return diceType;
//    }
//
//    public void setStateExpanded(boolean stateExpanded) {
//        this.stateExpanded = stateExpanded;
//    }
//
//    @Override
//    public int getLayoutRes() {
//        return R.layout.dice_result_item;
//    }
//
//    @Override
//    public SingleDiceTypeResultViewHolder createViewHolder(FlexibleAdapter adapter, LayoutInflater inflater, ViewGroup parent) {
//        return new SingleDiceTypeResultViewHolder(inflater.inflate(getLayoutRes(), parent, false), adapter);
//    }
//
//    @Override
//    public void bindViewHolder(FlexibleAdapter adapter, SingleDiceTypeResultViewHolder holder, int position, List payloads) {
//        if (adapter instanceof DiceAdapter) {
//            holder.bind(this, ((DiceAdapter) adapter).getDicePresenter());
//        } else {
//            throw new IllegalArgumentException("no DiceAdapter");
//        }
//    }
//
//    @Override
//    public boolean equals(Object o) {
//        if (this == o) return true;
//        if (o == null || getClass() != o.getClass()) return false;
//
//        SingleDiceTypeResultViewModel that = (SingleDiceTypeResultViewModel) o;
//
//        if (totalResultValue != that.totalResultValue) return false;
//        return diceResults != null ? diceResults.equals(that.diceResults) : that.diceResults == null;
//    }
//
//    @Override
//    public int hashCode() {
//        int result = diceResults != null ? diceResults.hashCode() : 0;
//        result = 31 * result + totalResultValue;
//        return result;
//    }
//}

File: app/src/main/java/com/alekseyvalyakin/roleplaysystem/ribs/game/active/dice/diceresult/SingleDiceTypeResultViewModel.kt
package com.alekseyvalyakin.roleplaysystem.ribs.game.active.dice.diceresult

import android.view.LayoutInflater
import android.view.ViewGroup

import com.alekseyvalyakin.roleplaysystem.flexible.FlexibleLayoutTypes
import com.alekseyvalyakin.roleplaysystem.ribs.game.active.dice.model.DiceResult
import com.alekseyvalyakin.roleplaysystem.ribs.game.active.dice.model.DiceType

import eu.davidea.flexibleadapter.FlexibleAdapter
import eu.davidea.flexibleadapter.items.AbstractFlexibleItem

data class SingleDiceTypeResultViewModel(
        val diceResults: List<DiceResult>,
        val totalResultValue: Int,
        val diceType: DiceType) : AbstractFlexibleItem<SingleDiceTypeResultViewHolder>() {

    override fun getLayoutRes(): Int {
        return FlexibleLayoutTypes.SINGLE_DICE_RESULT
    }

    override fun createViewHolder(adapter: FlexibleAdapter<*>, inflater: LayoutInflater, parent: ViewGroup): SingleDiceTypeResultViewHolder {
        return SingleDiceTypeResultViewHolder(SingleDiceTypeResultView(parent.context), adapter)
    }

    override fun bindViewHolder(adapter: FlexibleAdapter<*>, holder: SingleDiceTypeResultViewHolder, position: Int, payloads: List<*>?) {
        if (adapter is DiceResultAdapter) {
            holder.bind(this, adapter.relay)
        } else {
            throw IllegalArgumentException("no DiceAdapter")
        }
    }
}

